python/tvm/relax/vm_build.py

```diff
@@ -297,6 +297,10 @@ def foo(x: Tensor((3, 4), "float32"), y: Tensor((3, 4), "float32")):
     passes.append(relax.transform.ToNonDataflow())
     passes.append(relax.transform.CallTIRRewrite())
     passes.append(relax.transform.StaticPlanBlockMemory())
+
+    if tvm.transform.PassContext.current().config.get("relax.backend.use_cuda_graph", False):
```

Comment:
As we want to keep the building pipeline minimum, I'm not sure if there is any other way to decouple cuda graph rewriting and the "minimum" building pipeline. I know it may depends on the static memory planning, so I won't block it but would like us to consider it carefully. Also cc @tqchen

Reply:
Perhaps one approach is to lift out the static memory planning and TIR rewrite phases into another function. 

Then cuda graph can call these passes from outside and then pass to build, which detects if the function have already pass the runtime optimization stage, and skips the passes